### PR TITLE
Feature  web3 provider

### PR DESCRIPTION
**Pull request by digitaldonkey**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "walletconnect-monorepo",
+  "version": "0.7.8",
```
Comment (pedrouid) [inline package.json]: This needs to be removed, this package.json is not a package!!!

I've told you this repository is a wrapper to manage the other packages. There is no such thing a `walletconnect-monorepo` on NPM and there shouldnt be. Please look into monorepos further!

File: packages/walletconnect-web3-subprovider/src/index.js [hunk 2]
```diff
@@ -88,7 +98,7 @@ export default class WalletConnectSubprovider extends Subprovider {
   }
   sendAsync(payload, callback) {
     const next = () => {
-      const sendAsync = this.engine.sendAsync.bind(this)
+      const sendAsync = this._walletconnect.sendAsync.bind(this)
```
Comment (pedrouid) [inline packages/walletconnect-web3-subprovider/src/index.js]: Are you still getting an issue with this issue? I've successfully triggered JSON-RPC requests on mobile using the `walletconnect-web3-provider` package using the README instructions. If this issue is related with Walleth specifically, I've contacted @ligi and he said that hasn't updated to the latest spec.

Comment (digitaldonkey) [reply packages/walletconnect-web3-subprovider/src/index.js]: Sorry. A leftover from #38

Comment (digitaldonkey) [reply packages/walletconnect-web3-subprovider/src/index.js]: The above engine vs _walletconnect I will check.

**Comment by digitaldonkey** [comment]
Now everything should work and be merged.
If it works with your previous code I'm pretty sure it will fail on a slower machine as session is not enforced but only available due to lucky timing.

File: packages/walletconnect/src/index.js [hunk 3]
```diff
@@ -167,7 +167,18 @@ export default class WalletConnect extends Connector {
     const result = await this._getEncryptedData(`/session/${this.sessionId}`)
 
     if (result) {
-      if (result.data.approved) {
+      if (result.data && !result.data.approved) {
```
Comment (digitaldonkey) [inline packages/walletconnect/src/index.js]: This part is just legacy to be able to use WallEth before upgrade. 
It's not important.

**Comment by pedrouid** [comment]
> Now everything should work and be merged.
> If it works with your previous code I'm pretty sure it will fail on a slower machine as session is not enforced but only available due to lucky timing.

The reason it currently works is because I changed to implementation as you can read the documentation was updated on the [README](https://github.com/WalletConnect/walletconnect-monorepo/blob/master/packages/walletconnect-web3-provider/README.md) for this package.

**Comment by digitaldonkey** [comment]
That might help 👍

What about a **fallback** until Liggi updates WallEth? 
My solution might be shitty, but it alt least allow me to use and test. 
https://github.com/WalletConnect/walletconnect-monorepo/pull/42/files#diff-a2d67c3933f19104ca576fd5707ca80f

Devcon is near and WalletConnect is currently only usable with iPhones - no Android supported.

**Comment by digitaldonkey** [comment]
You might also want to add a try/catch wrapper to the readme. 

At some point the session times out and the user is left without any feedback.

https://github.com/digitaldonkey/web3multisign/blob/65406eeb5aedcc9907b5ed7df8fc1589311843dd/src/components/signerDialogs/WalletConnect.vue#L117-L128

It would be also nice if the arguments on would be respected and carried on.
I've never seen the defaults used, but at least I can't change interval and timeout. 
```listenSessionStatus(interval, timeout)```

**Comment by pedrouid** [comment]
> That might help 👍
> 
> What about a **fallback** until Liggi updates WallEth?
> My solution might be shitty, but it alt least allow me to use and test.
> https://github.com/WalletConnect/walletconnect-monorepo/pull/42/files#diff-a2d67c3933f19104ca576fd5707ca80f
> 
> Devcon is near and WalletConnect is currently only usable with iPhones - no Android supported.

I'm going to work with Ligi this weekend on upgrading to the latest spec so it will be ready before DevCon



> You might also want to add a try/catch wrapper to the readme.
> 
> At some point the session times out and the user is left without any feedback.
> 
> https://github.com/digitaldonkey/web3multisign/blob/65406eeb5aedcc9907b5ed7df8fc1589311843dd/src/components/signerDialogs/WalletConnect.vue#L117-L128
> 
> It would be also nice if the arguments on would be respected and carried on.
> I've never seen the defaults used, but at least I can't change interval and timeout.
> `listenSessionStatus(interval, timeout)`

That's a good point! I will added it directly to the Listener class. Also the defaults are available there:
https://github.com/WalletConnect/walletconnect-monorepo/blob/master/packages/walletconnect-core/src/listener.js#L19

**Comment by digitaldonkey** [comment]
Timeout of 6 sec? I don't think it's true.

It would be just great if I can pass poll interval and timeout on with listenSessionStatus(). 
You don't need to document it, but it would be great if you can override it.

**Comment by pedrouid** [comment]
I see what you mean, that's my bad! You can override from `walletconnect` library but forgot to pass it on `walletconnect-web3-provider`, I will push an update in a second

**Comment by W3stside** [comment]
Hi guys - popping in here without without a ton of background info or context on this PR but does this perhaps fix the `There was a problem connection to Example Dapp` issue? 

We (Gnosis) are currently integrating you guys and I am running into this issue using WallETH on Android. 

If I'm completely wrong just ignore me and continue on. Nice work anyways, seems really promising! :)

**Comment by pedrouid** [comment]
Both me and Ligi got carried away with all the events happening in Prague, so currently WallETH is still on WalletConnect v0.5.x while the Example Dapp is on v0.7.x. Alternatively you can try Balance Wallet on iOS which has published an update today for v0.7.x.

**Comment by pedrouid** [comment]
Several changes have been handled with v0.7.x changes on Example Dapp, WallETH, Balance Wallet and Tokenary Wallet. Web3 Provider future changes will re-visited with Issue #50

**Pull request closed without merging**